bundles/org.openhab.binding.asuswrt/src/main/java/org/openhab/binding/asuswrt/internal/structures/AsuswrtClientList.java

```diff
@@ -0,0 +1,200 @@
+            jsonList.remove(JSON_MEMBER_MACLIST);
+            jsonList.remove(JSON_MEMBER_API_LEVEL);
+            /* iterate MAC-Addresslist */
+            jsonList.keySet().forEach(macAddress -> {
+                if (isValidMacAddress(macAddress)) {
+                    AsuswrtClientInfo clientInfo = new AsuswrtClientInfo(jsonList.getAsJsonObject(macAddress));
+                    addClient(clientInfo);
+                } else {
+                    logger.trace("getClientlist: {} '{}'", ERR_INVALID_MAC_ADDRESS, macAddress);
+                }
+            });
+        } catch (Exception e) {
+            logger.debug("getClientlist: {} - {}'", ERR_JSON_FOMRAT, e.getMessage());
+        }
+    }
+
+    /**
+     * ADD CLIENT TO LIST
+     * 
+     * @param clientInfo AsuswrtClientInfo
+     */
+    private void addClient(AsuswrtClientInfo clientInfo) {
+        this.clientList.add(clientInfo);
+    }
+
+    /***********************************
+     *
+     * GET VALUES
+     *
+     ************************************/
+
+    /**
+     * GET CLIENT BY NAME
+     * 
+     * @param clientName String clientName
+     * @return AsuswrtClientInfo
+     */
+    public AsuswrtClientInfo getClientByName(String clientName) {
+        for (AsuswrtClientInfo client : this.clientList) {
+            if (client.getName().equals(clientName)) {
+                return client;
+            }
+        }
+        return new AsuswrtClientInfo();
+    }
+
+    /**
+     * GET CLIENT BY MAC
+     * 
+     * @param clientMAC String client MAC-Address
+     * @return AsuswrtClientInfo
+     */
+    public AsuswrtClientInfo getClientByMAC(String clientMAC) {
+        for (AsuswrtClientInfo client : this.clientList) {
+            if (client.getMac().equals(clientMAC)) {
+                return client;
+            }
+        }
+        return new AsuswrtClientInfo();
+    }
+
+    /**
+     * GET CLIENT BY IP
+     * 
+     * @param clientMAC String client IP-Address
+     * @return AsuswrtClientInfo
+     */
+    public AsuswrtClientInfo getClientByIP(String clientIP) {
+        for (AsuswrtClientInfo client : this.clientList) {
+            if (client.getIP().equals(clientIP)) {
+                return client;
+            }
+        }
+        return new AsuswrtClientInfo();
+    }
+
+    /*
+     * Return ; seperated list with clientNames and macAddresses
+     */
+    public String getClientList() {
+        StringBuilder clients = new StringBuilder();
+        for (AsuswrtClientInfo client : this.clientList) {
+            clients.append(client.getName() + " [" + client.getMac() + "]; ");
+        }
+        return clients.toString();
+    }
+
+    /*
+     * Return ; seperated list with clientNames
+     */
+    public String getClientNames() {
+        StringBuilder clients = new StringBuilder();
+        for (AsuswrtClientInfo client : this.clientList) {
+            clients.append(client.getName() + "; ");
+        }
+        return clients.toString();
```

Comment:
You can use Streams and turn this into a one line:

```suggestion
        return clientList.stream().map(AsuswrtClientInfo::getName).collect(Collectors.joining("; "));
```